modules/visual_script/visual_script_editor.cpp

```diff
@@ -1167,6 +1167,156 @@ void VisualScriptEditor::_update_members() {
 	updating_members = false;
 }
 
+String VisualScriptEditor::_sanitized_variant_text(const StringName &property_name) {
+	Variant var = script->get_variable_default_value(property_name);
+
+	switch (script->get_variable_info(property_name).type) {
+		case Variant::Type::NIL:
+			var = Variant((Variant)var);
```

Comment:
I'm surprised Variant doesn't have a method to cast to a given `Variant::Type`. It already has `can_convert`, but if it can convert, then the conversion has to be done with a very long switch statement if it could be anything... @reduz Maybe this should be added to Variant? I wonder if there is any similar code like this that could be replaced.

Reply:
There are plenty of switch statements that depend on Variant's type, but not so much one to just convert a Variant from one to another. You can just stick a Quat inside a Variant if you need a Variant that's a quat, after all. At least, not that I could find with a cursory search.

This is only really an issue for the visual script editor not updating the label, because the internal Variant only changes type when an appropriate value is used. For any other uses, just using the actual type directly, either in C++ or with typed GDScript, is the standard modus operandi.

Not sure if we need a `convert` function (but I was still surprised not to find one.)

Reply:
There is still something of a convert function. Here's how I fixed this bug in my version, I basically added something like this to convert the value of the variant to the correct type so it got stringified correctly:

```
if (pinfo.type != Variant::NIL && var.get_type() != pinfo.type) {
	Callable::CallError ce;
	const Variant *var_p= &var;
	var = Variant::construct(pinfo.type, &var_p, 1, ce, false);
}
```

Reply:
That's much nicer code. Thanks!